app/services/images/delete.rb

```diff
@@ -0,0 +1,25 @@
+# app/services/images/delete.rb
+
+module Images
+  class Delete
+    attr_accessor :image_paths
+
+    def initialize(image_paths)
+      @image_paths = image_paths
+    end
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def call
+      uploader = ArticleImageUploader.new
+      image_paths.each do |image_path|
+        next if image_path.blank?
+
+        uploader.retrieve_from_store!(image_path)
+        uploader.remove!
```

Comment:
Do we need to retrieve the object before we remove it?

Reply:
I can attest that this works since I [recommended it ](https://github.com/forem/forem/issues/13936#issuecomment-872514868)and tested it on Staging. I found suggestion [here](https://groups.google.com/g/carrierwave/c/15ALbubO5Uk).

Reply:
@maestromac I have no doubt it works. I did not mean to imply it wouldn't. 🙂 I mean do we _need_ to fetch the file before deleting it or can we simply delete it? For example, we can delete a file on disk if we know its path without reading it. If the file is in remote storage, we can issue a `DELETE` request without preceding it with a `GET` request.

Retrieving the file feels like unnecessary overhead here, so I'm wondering if CarrierWave requires that we do that or if we can skip that step.

Reply:
After diving a bit into CarrierWave's API. I can't seem to find a way to do it without retrieving the file first :confused: